src/components/ErrorBoundary/ErrorBoundary.js

```diff
@@ -27,18 +31,65 @@ class ErrorBoundary extends React.Component<Props, State> {
         onError: noop,
     };
 
-    componentDidCatch(error: Error): void {
+    componentDidCatch(error: Error, info: Object): void {
         this.setState({ error }, () => {
-            this.props.onError(error);
+            this.handleError(
+                error,
+                ERROR_CODE_UNEXPECTED_EXCEPTION,
+                true,
+                {
+                    ...info,
+                },
+                this.props.errorOrigin,
+            );
         });
     }
 
+    /**
+     * Formats the error and emits it to the top level onError prop
+     *
+     * @param {Error} e - the error which occured
+     * @param {string} type - the error type to identify where the error occured
+     * @param {string} code - the error code to identify what error occured
+     * @param {Object} contextInfo - additional information which may be useful for the consumer of the error
+     * @return {void}
+     */
+    handleError = (
+        error: ElementsXhrError,
+        code: string,
+        isErrorDisplayed: boolean = false,
```

Comment:
Does this need to be a separate param if it's being injected into `context_info` anyway? Seems like it could move into the object that's passed in.